src/cascadia/TerminalSettingsEditor/SettingContainer.cpp

```diff
@@ -0,0 +1,114 @@
+        if (auto resetButton{ obj.GetTemplateChild(L"ResetButton") })
+        {
+            if (auto elem{ resetButton.try_as<UIElement>() })
+            {
+                elem.Visibility(newVal ? Visibility::Visible : Visibility::Collapsed);
+            }
+        }
+
+        // update visibility for override message
+        if (auto overrideMsg{ obj.GetTemplateChild(L"OverrideMessage") })
+        {
+            if (auto elem{ overrideMsg.try_as<UIElement>() })
+            {
+                elem.Visibility(newVal ? Visibility::Visible : Visibility::Collapsed);
+            }
+        }
+    }
+
+    void SettingContainer::OnApplyTemplate()
+    {
+        if (auto child{ GetTemplateChild(L"ResetButton") })
+        {
+            if (auto button{ child.try_as<Controls::Button>() })
+            {
+                // Apply click handler for the reset button.
+                // When clicked, we dispatch the bound ClearSettingValue event,
+                // resulting in inheriting the setting value from the parent.
+                button.Click([=](auto&&, auto&&) {
+                    _ClearSettingValueHandlers(*this, nullptr);
+                });
+
+                // apply tooltip and help text (automation property)
+                const auto& helpText{ RS_(L"SettingContainer_ResetButtonHelpText") };
+                Controls::ToolTipService::SetToolTip(child, box_value(helpText));
+                Automation::AutomationProperties::SetName(child, helpText);
+
+                // initialize visibility for reset button
+                button.Visibility(HasSettingValue() ? Visibility::Visible : Visibility::Collapsed);
+            }
+        }
+
+        if (auto child{ GetTemplateChild(L"OverrideMessage") })
+        {
+            if (auto tb{ child.try_as<Controls::TextBlock>() })
+            {
+                // initialize visibility for reset button
+                tb.Visibility(HasSettingValue() ? Visibility::Visible : Visibility::Collapsed);
+            }
+        }
+
+        if (auto content{ Content() })
+        {
+            if (auto obj{ content.try_as<DependencyObject>() })
+            {
+                // apply header text as name (automation property)
+                if (const auto& header{ Header() })
+                {
+                    const auto headerText{ header.try_as<hstring>() };
+                    if (headerText && !headerText->empty())
+                    {
+                        Automation::AutomationProperties::SetName(obj, *headerText);
+                    }
+                }
+
+                // apply help text as tooltip and help text (automation property)
+                const auto& helpText{ HelpText() };
+                if (!helpText.empty())
+                {
+                    Controls::ToolTipService::SetToolTip(obj, box_value(helpText));
+                    Automation::AutomationProperties::SetHelpText(obj, helpText);
+                }

```

Comment:
surprised you couldn't do some binding work to make this happen actually. You should be able to `{TemplateBinding HelpText}` inside the xaml file. Don't block on this. Just curious.

Reply:
Same for `{TemplateBinding Header}` on automation property